test/test_images/singlethreaded/main.go

```diff
@@ -27,18 +27,16 @@ import (
 	"knative.dev/serving/test"
 )
 
-var lockedFlag = atomic.NewInt32(-1)
+var isLocked = atomic.NewBool(false)
 
 func handler(w http.ResponseWriter, r *http.Request) {
-	v := lockedFlag.Inc() // Returns the new value.
-	defer lockedFlag.Dec()
-	if v > 0 {
-		// Return HTTP 500 if more than 1 request at a time gets in
-		w.WriteHeader(http.StatusInternalServerError)
-		return
+	if isLocked.CAS(false /*was unlocked*/, true /*lock*/) {
+		defer isLocked.Store(false)
+		time.Sleep(500 * time.Millisecond)
+		fmt.Fprintf(w, "One at a time")
```

Comment:
Should probably `return` here to avoid doing a superfluous WriteHeader(500) in the success case (tho looks like [the only effect otherwise might be a log message](https://github.com/golang/go/blob/520f3b72db7befab2028d9a47376267cf2d274a9/src/net/http/server.go#L1143) 🤷‍♂️)

Reply:
Done